### ISSUE TITLE
Hiding GLFW window from dock with LSUIElement=1

### ISSUE THREAD

**Issue by wormyrocks** [glfw/glfw#1552]
GLFW 3.3
macOS 10.15 beta (but seeing the same behavior on Mojave)

Hi all!
Been working on a crossplatform app for a while that keeps its window hidden most of the time and needs to be as invisible as possible when it is in that state. It gets opened by an RPC call from an external process.
On Windows, I use [this](https://github.com/glfw/glfw/issues/293#issuecomment-336331610) to pop up the window as a tool window, but it seems like there's no good equivalent on macOS, even when all the app windows are hidden. I tried following [tutorials](https://apple.stackexchange.com/questions/92004/is-there-a-way-to-hide-certain-apps-from-the-cmdtab-menu) to bundle the executable produced as a .app package with a switch in Info.plist, but none of them seem to work (when I double click the .app it still pops up in my dock) and I'm not sure how to go about debugging.
Does GLFW override this behavior somewhere in the Objective-C glue code and is there possibly a way I could patch to revert it, or some flag I might be able to set with something returned from glfwGetCocoaWindow? Have other people needed to do this and had success in the past?

Thanks!

**Comment by seaside-wu** [comment]
I have the same problem. Does anyone know the solution?

**Comment by wormyrocks** [comment]
Yep, figured it out. In cocoa_window.m in the createNativeWindow() function, change:
```
[NSApp setActivationPolicy:NSApplicationActivationPolicyRegular];
```
to: 
```
[NSApp setActivationPolicy:NSApplicationActivationPolicyAccessory];
```

**Comment by Chlumsky** [comment]
If anyone is wondering how to achieve this without modifying GLFW, I managed to do it by adding the following MM file, which spoofs the `setActivationPolicy:` method called in GLFW initialization:

```mm
#import <AppKit/AppKit.h>
#import <objc/runtime.h>

@interface NSApplication (GLFW_CLI_Application)
- (BOOL) setActivationPolicyToProhibited: (NSApplicationActivationPolicy) activationPolicy;
@end

@implementation NSApplication (GLFW_CLI_Application)

+ (void) load {
    Method original = class_getInstanceMethod(self, @selector(setActivationPolicy:));
    Method swizzled = class_getInstanceMethod(self, @selector(setActivationPolicyToProhibited:));
    method_exchangeImplementations(original, swizzled);
}

- (BOOL) setActivationPolicyToProhibited: (NSApplicationActivationPolicy) activationPolicy {
    // The following will actually call the original setActivationPolicy: method because they were switched
    [self setActivationPolicyToProhibited: NSApplicationActivationPolicyProhibited];
}

@end
```

My use case was a purely command line program that uses OpenGL, so you may want to use `NSApplicationActivationPolicyAccessory` instead of `NSApplicationActivationPolicyProhibited`.
